app/src/main/java/com/mychefassistant/presentation/ingredient/IngredientViewModel.kt

```diff
@@ -1,21 +1,35 @@
 package com.mychefassistant.presentation.ingredient
 
-import android.app.Application
-import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
 import com.mychefassistant.core.domain.Kitchen
 import com.mychefassistant.framework.ChefAssistantViewModel
-import com.mychefassistant.framework.Interactors
-import kotlinx.coroutines.GlobalScope
+import com.mychefassistant.framework.interactors.KitchenInteractors
+import com.mychefassistant.utils.Event
 import kotlinx.coroutines.launch
 
-class IngredientViewModel(application: Application, interactors: Interactors) :
-    ChefAssistantViewModel(application, interactors) {
+class IngredientViewModel(private val kitchenInteractors: KitchenInteractors) :
+    ChefAssistantViewModel() {
+    private var kitchenId = 0
+    lateinit var kitchen: Kitchen
 
-    val kitchen: MutableLiveData<Kitchen> = MutableLiveData()
+    fun setKitchenId(id: Int) {

```

Comment:
Find another way of sending this. The function is redundant.
This can be received in the start function.